astropy/io/fits/tests/test_core.py

```diff
@@ -514,15 +514,15 @@ def test_consecutive_writeto(self):
             # Add a bunch of header keywords so that the data will be forced to
             # new offsets within the file:
             for idx in range(40):
-                hdul1[1].header['TEST%d' % idx] = 'test'
+                hdul1[1].header['TEST{}'.format(idx)] = 'test'
 
             hdul1.writeto(self.temp('test1.fits'))
             hdul1.writeto(self.temp('test2.fits'))
 
             # Open a second handle to the original file and compare it to hdul1
             # (We only compare part of the one header that was modified)
             # Compare also with the second writeto output
-            with fits.open(self.data('test0.fits')) as hdul2:
+            with fits.open(self.data('test0.fits')) as hdu2:
```

Comment:
change was due to big red coloring of the editor for the unused reference, but you're right, and now reading the comments it seems that the test wasn't doing the right thing anyway